staging/src/k8s.io/apimachinery/pkg/watch/streamwatcher_test.go

```diff
@@ -20,32 +20,66 @@ import (
 	"fmt"
 	"io"
 	"reflect"
+	goruntime "runtime"
+	"sync"
 	"testing"
+	"time"
 
 	"k8s.io/apimachinery/pkg/runtime"
 	. "k8s.io/apimachinery/pkg/watch"
 )
 
 type fakeDecoder struct {
-	items chan Event
-	err   error
+	lock   sync.Mutex
+	items  chan Event
+	err    error
+	count  int
+	closed int
```

Comment:
Use a bool?

Reply:
I need to check whether it is called twice later in the race check